libraries/AP_Compass/AP_Compass.cpp

```diff
@@ -788,8 +788,8 @@ void Compass::_detect_backends(void)
 #elif HAL_COMPASS_DEFAULT == HAL_COMPASS_AERO
     ADD_BACKEND(DRIVER_BMM150, AP_Compass_BMM150::probe(*this, hal.i2c_mgr->get_device(HAL_COMPASS_BMM150_I2C_BUS, HAL_COMPASS_BMM150_I2C_ADDR)),
                 AP_Compass_BMM150::name, false);
-    ADD_BACKEND(DRIVER_HMC5883, AP_Compass_HMC5843::probe(*this, hal.i2c_mgr->get_device(HAL_COMPASS_HMC5843_I2C_BUS, HAL_COMPASS_HMC5843_I2C_ADDR), true),
-                AP_Compass_HMC5843::name, true);
+ //   ADD_BACKEND(DRIVER_HMC5883, AP_Compass_HMC5843::probe(*this, hal.i2c_mgr->get_device(HAL_COMPASS_HMC5843_I2C_BUS, HAL_COMPASS_HMC5843_I2C_ADDR), true),
```

Comment:
why is this driver being removed?

Reply:
it should not, how can I get rid of this file modification in the current PR?

Reply:
You seem like you know a little about git so I'll jump ahead and see if you can fix it yourself. If this gets too complicated then we can help you out.

Since your commit messages don't quite agree with the coding style, I'd suggest:
- reset MIXED (not hard)
- remove the lines you don't want committed
- recommit the changes grouped in each directory per the style guide